python/pyspark/sql/tests/test_connect_compatibility.py

```diff
@@ -23,155 +23,165 @@
-            classic_methods = self.get_public_methods(classic_cls)
-            connect_methods = self.get_public_methods(connect_cls)
-
-            # Identify missing methods
-            classic_only_methods = set(classic_methods.keys()) - set(connect_methods.keys())
-
-            # Compare the actual missing methods with the expected ones
-            self.assertEqual(
-                classic_only_methods,
-                expected_missing_methods,
-                f"{cls_name}: Unexpected missing methods in Connect: {classic_only_methods}",
-            )
-
-        # Expected missing methods for DataFrame
-        expected_missing_methods_for_dataframe = {
-            "inputFiles",
-            "isLocal",
-            "semanticHash",
-            "isEmpty",
-        }
-
-        # DataFrame missing method check
-        check_missing_methods(
-            ClassicDataFrame, ConnectDataFrame, "DataFrame", expected_missing_methods_for_dataframe
+    def check_compatibility(
+        self,
+        classic_cls,
+        connect_cls,
+        cls_name,
+        expected_missing_properties,
+        expected_missing_methods,
+    ):
+        """
+        Main method for checking compatibility between classic and connect.
+
+        This method performs the following checks:
+        - API signature comparison between classic and connect classes.
+        - Property comparison, identifying any missing properties in the connect class.
+        - Method comparison, identifying any missing methods in the connect class.
+
+        Parameters
+        ----------
+        classic_cls : type
+            The classic class to compare.
+        connect_cls : type
+            The connect class to compare.
+        cls_name : str
+            The name of the class.
+        expected_missing_properties : set
+            A set of properties expected to be missing in the connect class.
+        expected_missing_methods : set
+            A set of methods expected to be missing in the connect class.
+        """
+        self._compare_method_signatures(classic_cls, connect_cls, cls_name)
+        self._compare_property_lists(
+            classic_cls, connect_cls, cls_name, expected_missing_properties
         )
+        self._check_missing_methods(classic_cls, connect_cls, cls_name, expected_missing_methods)
 
-        # Expected missing methods for Column (if any, replace with actual values)
-        expected_missing_methods_for_column = set()
-
-        # Column missing method check
-        check_missing_methods(
-            ClassicColumn, ConnectColumn, "Column", expected_missing_methods_for_column
+    def test_dataframe_compatibility(self):
+        """Test DataFrame compatibility between classic and connect."""
+        expected_missing_properties = {"sql_ctx", "isStreaming"}
+        expected_missing_methods = {"inputFiles", "isLocal", "semanticHash", "isEmpty"}
```

Comment:
Why are all of these (execpt for `sql_ctx`) not supported?

Reply:
We should make it a point to document every exception here.

Reply:
Just realized that there is a bug in test functions. They all should be supported.
Let me fix the test, thanks for catching this out!

Reply:
Fixed!